component-runtime-manager/src/test/java/org/talend/sdk/component/runtime/manager/service/LocalCacheServiceTest.java

```diff
@@ -15,46 +15,167 @@
+        }
+    };
+
+    private final LocalCacheService<String> cache = new LocalCacheService<>("LocalCacheServiceTest");
+
+
+    @BeforeEach
+    void init() {
+        this.active = false;
+        this.intervale = 0;
+        final Map<Class<?>, Object> services = new HashMap<>(2);
+       // services.put(LocalCache.class, new LocalCacheService("LocalCacheServiceTest"));
+        services.put(LocalConfiguration.class,
+                new LocalConfigurationService(Collections.singletonList(cacheConfig), "test"));
+
+
+        final PropertyEditorRegistry propertyEditorRegistry = new PropertyEditorRegistry();
+        Injector injector = new InjectorImpl("LocalCacheServiceTest",
+                new ReflectionService(new ParameterModelService(propertyEditorRegistry), propertyEditorRegistry),
+                new ProxyGenerator(), services);
+
+        injector.inject(cache);
+        Assertions.assertTrue(cache.isEmpty(), "not empty after creation");
+    }
+
+    @AfterEach
+    void end() {
+        cache.clean();
+    }
+
     @Test
     void conditionalEviction() {
-        final LocalCache cache = new LocalCacheService("LocalCacheServiceTest");
-        final String value = cache.computeIfAbsent("foo", () -> "bar");
+        final String value = this.cache.computeIfAbsent("foo", () -> "bar");
         assertEquals("bar", value);
         assertEquals("bar", cache.computeIfAbsent("foo", () -> "test"));
+
         cache.evictIfValue("foo", value);
         assertEquals("renewed", cache.computeIfAbsent("foo", () -> "renewed"));
+
+        cache.evictIfValue("foo", "doNothing");
+        assertEquals("renewed", cache.computeIfAbsent("foo", () -> "renewed"));
+
+        cache.evictIfValue("foo", null);
+        assertEquals("renewed", cache.computeIfAbsent("foo", () -> "renewed"));
     }
 
     @Test
     void eviction() {
-        final LocalCache cache = new LocalCacheService("LocalCacheServiceTest");
         assertEquals("bar", cache.computeIfAbsent("foo", () -> "bar"));
         cache.evict("foo");
         assertEquals("renewed", cache.computeIfAbsent("foo", () -> "renewed"));
     }
 
+    @Test
+    void mustRemoved() {
+        final boolean[] mustRemoved = new boolean[] { false };
+
+        final String v1 = cache.computeIfAbsent("key1", (Element<String> e) -> mustRemoved[0], () -> "value1");
+        final String v2 = cache.computeIfAbsent("key1", () -> "value2");
+        mustRemoved[0] = true;
+        final String v3 = cache.computeIfAbsent("key1", () -> "value3");
+        assertEquals("value1", v1);
+        assertEquals("value1", v2);
+        assertEquals("value3", v3);
+    }
+
+    @Test
+    void localTimeout() throws InterruptedException {
+        final String v1 = cache.computeIfAbsent("key1", 200L, () -> "value1");
+        final String v2 = cache.computeIfAbsent("key1", 200L,  () -> "value2");
+
+        final String valueKey2 = cache.computeIfAbsent("key2", -1L, () -> "valueKey2");
+
+        Thread.sleep(240L);
```

Comment:
I'd increase a bit the latency, windows is bad on that so likely +-100ms is not luxury (you can increase the timeout (200) if you judge the % of the error is too close to the value)